src/plugins/intel_gna/optimizer/gna_pass_manager.cpp

```diff
@@ -978,6 +979,33 @@ void InsertCopyLayerPass::run() {
             }
         }
     }
+
+    for (auto & l : *pLayers) {
+        if (!l->insData.empty()) continue;
+
+        for (auto output : l->outData) {
+            auto& inputTo = getInputTo(output);
+            for (auto& childLayer : inputTo) {
+                auto current_layer = childLayer.second;
+                auto previous_layer = l;
+                std::vector<int> connections = CNNLayerFindInsDataIdxes(output, childLayer.second);
+
+                for (auto input_idx : connections) {
+                    while (LayerInfo(current_layer).isNonFunctional()) {
+                        if (current_layer->outData.size() == 0) break;
+                        if (getInputTo(current_layer->outData[0]).size() == 0) break;
+                        previous_layer = current_layer;
+                        current_layer = CNNNetGetNextLayerSkipCertain(current_layer, 0, input_idx, [](CNNLayerPtr origin){return false;}).first;
+                    }
+                    // exit by a while condition -> we found functional layer
+                    if (!LayerInfo(current_layer).isNonFunctional()) continue;
```

Comment:
isNonFunctional() will return false for split and concat but they also can be a part of non-functional subgraphs

Reply:
I will extend the test coverage to check this